### PR TITLE
USHIFT-126: Minimal CSI Plugin documentation

### PR DESCRIPTION
**Pull request by copejon**
This PR adds some minimal documentation describing the default storage plugin, prerequisites, and provides an example workload.

**Which issue(s) this PR addresses**:
Closes [USHIFT-126](https://issues.redhat.com//browse/USHIFT-126)

### CODE REVIEW

File: docs/default_csi_plugin.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# MicroShift Default CSI Plugin
+
+MicroShift enables dynamic storage provisioning out of the box with the ODF-LVM CSI plugin. This plugin is downstream
+Red Hat distribution of TopoLVM. This feature is not currently configurable but will be in the near future. For more
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:4]: Don't sell futures. Talk about what's possible now, and leave the rest for future PRs.

File: docs/default_csi_plugin.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# MicroShift Default CSI Plugin
+
+MicroShift enables dynamic storage provisioning out of the box with the ODF-LVM CSI plugin. This plugin is downstream
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:3]: It's implied in the name, but you could be explicit here and say that ODF-LVM provides a separate logical volume for each PersistentVolumeClaim.

File: docs/default_csi_plugin.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Design
+
+ODF-LVM manifests can be found in [microshift/assets/components/odf-lvm](../assets/components/odf-lvm). For an overview
+of ODF-LVM architecture, see the [design doc](https://github.com/red-hat-storage/topolvm/blob/main/docs/design.md).
+
+## Deployment
+
+ODF-LVM is deployed onto the cluster in the `openshift-storage` namespace, after MicroShift boots. In order to reduce
+MicroShift's storage footprint and cluster overhead,
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:15]: This statement about capacity tracking would fit in well in the enhancement document to explain the choice. Here I think it's simplest to focus on the outcome from the user's perspective: "ODF-LVM uses StorageCapacity tracking to ensure that pods are not schedule if the storage they request is not available" (I think that's what it does?)

File: docs/default_csi_plugin.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+[StorageCapacity tracking](https://kubernetes.io/docs/concepts/storage/storage-capacity/) is used in lieu of the
+plugin's extended-scheduler.
+
+## System Requirements
+
+### Volume Group Name
+
+LVMD, a component of ODF-LVM, requires that volume groups specified in
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:23]: Users don't have direct influence to configure the sub-components separately, right? So we can just talk about the CSI driver as a whole thing.

```suggestion
ODF-LVM requires that volume groups specified in
```

File: docs/default_csi_plugin.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+the [device-class configuration](../assets/components/odf-lvm/topolvm-lvmd-config_configmap_v1.yaml) exist at the
+runtime. If the volume group doesn't exist, the LVMD container will fail to start and enter a CrashLoopBackoff state.
+The initial integration of ODF-LVM assumes volume-group named `rhel` exists. Unallocated space must exist on the volume
+group if the plugin is to dynamically provision logical volumes on it.
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:27]: I think these last few sentences can be combined a bit to bring the specific information earlier. Yes, we require that the VG in the device-class exist, but further the name must be `rhel` right now and we should lead with that. When it's configurable, then we can talk about what the default might be (if we have a default), but for now we need to make it clear that MicroShift must be deployed in such a way that there is a VG with that exact name.

File: docs/default_csi_plugin.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+```shell
+  Warning  ProvisioningFailed    3s (x2 over 5s)  topolvm.cybozu.
+  com_topolvm-controller-858c78d96c-xttzp_0fa83aef-2070-4ae2-bcb9-163f818dcd9f failed to provision volume with 
+  StorageClass "topolvm-provisioner": rpc error: code = ResourceExhausted desc = no enough space left on VG: 
+  free=(BYTES_INT), requested=(BYTES_INT)
+```
+
+## Usage
+
+ODF-LVM's [StorageClass](../assets/components/odf-lvm/topolvm_default-storage-class.yaml) is deployed by with a default
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:43]:
```suggestion
ODF-LVM's [StorageClass](../assets/components/odf-lvm/topolvm_default-storage-class.yaml) is deployed with a default
```

File: docs/default_csi_plugin.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+LVMD, a component of ODF-LVM, requires that volume groups specified in
+the [device-class configuration](../assets/components/odf-lvm/topolvm-lvmd-config_configmap_v1.yaml) exist at the
+runtime. If the volume group doesn't exist, the LVMD container will fail to start and enter a CrashLoopBackoff state.
+The initial integration of ODF-LVM assumes volume-group named `rhel` exists. Unallocated space must exist on the volume
+group if the plugin is to dynamically provision logical volumes on it.
+
+### Volume Size Increments
+
+ODF-LVM provisions storage in increments of 1Gb. Storage requests are rounded up to the near gigabyte. When a volume
```
Comment (dhellmann) [inline docs/default_csi_plugin.md:31]: Is it Gb or Gi?

File: docs/devenv_rhel8.md [hunk 8]
```diff
@@ -24,7 +24,15 @@ virt-install \
 
 In the OS installation wizard, set the following options:
 - Root password and `microshift` administrator user
-- In the Installation Destination, select automatic partitioning on the disk without encryption
+- Select Installation Destination
```
Comment (ggiguash) [inline docs/devenv_rhel8.md:27]: Nice! I copied this to my #826 PR

File: docs/default_csi_plugin.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
+# MicroShift Default CSI Plugin
```
Comment (ggiguash) [inline docs/default_csi_plugin.md:1]: Please, link this doc from the top-level README under developer section.

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: docs/default_csi_plugin.md (hunk 7 above)
Comment (copejon) [reply docs/default_csi_plugin.md:30]: Changed the PVC request suffix to mirror the text here ( `Gi` -> `G`)

**Pull request merged**